tests/messages/consumed/test_record.py

```diff
@@ -0,0 +1,80 @@
+from typing import Any
+from unittest.mock import patch
+import pytest
+from tol_lab_share.messages.consumed.record import Record
+
+
+@pytest.fixture
+def transform():
+    def decode_transform(v):
+        v.decode()
+
+    return decode_transform
+
+
+@pytest.fixture
+def payload() -> dict[str, Any]:
+    return {}
+
+
+class TestRecord:
+    @pytest.mark.parametrize(
+        "payload, name, parent_path",
+        [
+            ({}, "", ""),
+            ({}, "", "root"),
+            ({}, "name", ""),
+            ({}, "name", "root"),
+        ],
+    )
+    def test_initialises_successfully_with_all_parameters(self, payload, name, parent_path):
+        record = Record(payload, name, parent_path)
+        assert record is not None
+
+    def test_initialises_successfully_with_no_name(self):
+        record = Record({}, parent_path="parent_path")
+        assert record is not None
```

Comment:
Do we need to assert here that `record.name` is None as well?

Reply:
I had made a conscious decision to avoid testing the inner management of data on this type, but I have changed my mind after you comment, so I've updated the tests.  There is no `record.name` but I have added expectations for what `record._payload` and `record._path` should be.